src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -362,6 +362,11 @@ def attributes(self) -> List[Attribute]:
             self.source_plan, SelectStatement
         ):
             self.source_plan._attributes = attributes
+            if (
```

Comment:
Looks good, but we'll need to add a comment with explanation here too.

Reply:
+1
At first glance the assignment seems backwards but the schema query for `SnowflakePlan` comes from the `SelectStatement` so once a describe query gets executed, the results are being propagated back down to the `SelectStatement`. Is there a possibility we need to propagate the metadata down to even more `SelectStatement`s?

Reply:
Yes, added.

Reply:
How about Afroz's question. Can we think of cases where the propagation can go even further down?

Reply:
Just noticed you addressed the question in the other thread. SGTM.